app/src/main/java/tech/relaycorp/letro/repository/GatewayRepository.kt

```diff
@@ -13,15 +14,26 @@ import tech.relaycorp.awaladroid.GatewayClient
 import tech.relaycorp.letro.data.GatewayAvailabilityDataModel
 import javax.inject.Inject
 import javax.inject.Singleton
+import kotlinx.coroutines.flow.Flow
+import tech.relaycorp.awaladroid.endpoint.FirstPartyEndpoint
+import tech.relaycorp.awaladroid.endpoint.InvalidThirdPartyEndpoint
+import tech.relaycorp.awaladroid.endpoint.PublicThirdPartyEndpoint
+import tech.relaycorp.letro.R
 
 @Singleton
-class GatewayRepository @Inject constructor(@ApplicationContext var context: Context) {
+class GatewayRepository @Inject constructor(
+    @ApplicationContext var context: Context,
+    private val preferencesDataStoreRepository: PreferencesDataStoreRepository,
+) {
+
+    private val gatewayScope = CoroutineScope(Dispatchers.IO)
 
     private val _gatewayAvailabilityDataModel: MutableStateFlow<GatewayAvailabilityDataModel> =
         MutableStateFlow(GatewayAvailabilityDataModel.Unknown)
     val gatewayAvailabilityDataModel: StateFlow<GatewayAvailabilityDataModel> get() = _gatewayAvailabilityDataModel
 
-    private val gatewayScope = CoroutineScope(Dispatchers.IO)
+    private val _firstPartyEndpointNodeId: Flow<String?> = preferencesDataStoreRepository.getFirstPartyEndpoint()
+    private val _thirdPartyEndpointNodeId: Flow<String?> = preferencesDataStoreRepository.getThirdPartyEndpoint()
```

Comment:
Following up on the comment I made on the `PreferencesDataStoreRepository`, I think these should be prefixed with `server` and we can omit the `Node` bit.

Reply:
@gnarea Can you please make a suggestion as I'm not sure how server applies to the FirstPartyEndpoint

Reply:
@gnarea Although I applied the previous suggestion, I don't agree with the naming. Here is my reasoning and please correct me if I'm wrong. FirstParty isn't from the server, ThirdParty sure but what's the benefit of adding "server" prefix.
Also, omitting "NodeId" is misleading. The reason for naming it "NodeId" is because I'm just saving the NodeId from the First and Third PartyEndpoints.

Reply:
Partially done. Feel free to open again if you strongly believe the "NodeId" should be omitted